### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by doncem**
test library update gotcha: null does not match to any

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -72,21 +85,27 @@ sourceSets {
...
+  dependencies {
+    dependencySet(group: 'com.fasterxml.jackson.core', version: '2.9.5') {
+      entry 'jackson-annotations'
+      entry 'jackson-core'
+      entry 'jackson-databind'
+    }
+    dependencySet(group: 'com.fasterxml.jackson.datatype', version: '2.9.5') {
+      entry 'jackson-datatype-jsr310'
+    }
+  }
```
Comment (rkondratowicz) [inline build.gradle]: can you explain this part?

File: checkstyle.xml [hunk 2]
```diff
@@ -54,13 +56,15 @@
             <property name="tokens" value="LITERAL_TRY, LITERAL_FINALLY, LITERAL_IF, LITERAL_ELSE, LITERAL_SWITCH"/>
         </module>
         <module name="NeedBraces"/>
-        <module name="LeftCurly">
-            <property name="maxLineLength" value="100"/>
```
Comment (lgonczar) [inline checkstyle.xml]: Why did you remove the limit? What's the default?

Comment (doncem) [reply checkstyle.xml]: I've only replaced the file with current config we have in recent project aka send-letter-service

Comment (doncem) [reply checkstyle.xml]: it has been deprecated (guessing since v8) http://checkstyle.sourceforge.net/config_blocks.html#LeftCurly

File: build.gradle [hunk 3]
```diff
@@ -1,20 +1,20 @@
 plugins {
   id 'application'
   id 'checkstyle'
-  id 'io.spring.dependency-management' version '1.0.3.RELEASE'
+  id 'io.spring.dependency-management' version '1.0.5.RELEASE'
   id 'org.flywaydb.flyway' version '5.0.7'
-  id 'org.springframework.boot' version '1.5.7.RELEASE'
+  id 'org.springframework.boot' version '1.5.12.RELEASE'
```
Comment (timja-kainos) [inline build.gradle]: spring boot 2 upgrade to come?

Comment (doncem) [reply build.gradle]: Yes. That's a different ball game left aside

File: src/test/java/uk/gov/hmcts/reform/draftstore/service/crypto/CryptoServiceTest.java [hunk 4]
```diff
@@ -43,4 +43,30 @@ public void should_throw_an_exception_when_wrong_secret_is_used() throws Excepti
         assertThat(thrown)
             .isInstanceOf(InvalidKeyException.class);
     }
+
+    @Test
+    public void should_throw_an_exception_when_empty_input_received() {
```
Comment (lgonczar) [inline src/test/java/uk/gov/hmcts/reform/draftstore/service/crypto/CryptoServiceTest.java]: It would be ideal to have two different tests - one for encrypting and one for decrypting. Not strong about it, so just to consider.

Comment (doncem) [reply src/test/java/uk/gov/hmcts/reform/draftstore/service/crypto/CryptoServiceTest.java]: it would make sense if those tests actually are responsible for encrypting/decrypting. but the sole purpose is to cover the branch of conditional exception throw disregarding the following code coherence
